src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -135,39 +143,71 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
 
                     if (part.GetType() == typeof(ContentPart) && partTypeName != contentTypePartDefinition.ContentTypeDefinition.Name)
                     {
+                        // Todo: var shapeType = partTypeName; for Placement fix PR #8935
                         var shapeType = context.DisplayType != "Detail" ? "ContentPart_" + context.DisplayType : "ContentPart";
 
                         var shapeResult = new ShapeResult(shapeType, ctx => ctx.ShapeFactory.CreateAsync(shapeType, () => new ValueTask<IShape>(new ZoneHolding(() => ctx.ShapeFactory.CreateAsync("Zone")))));
                         shapeResult.Differentiator(partName);
+                        shapeResult.Name(partName);
                         shapeResult.Location("Content");
 
-                        await shapeResult.ApplyAsync(context);
+                        shapeResult.Displaying(ctx =>
+                        {
+                            string[] displayTypes = new[] { "", "_" + ctx.Shape.Metadata.DisplayType };
 
-                        var contentPartShape = shapeResult.Shape;
+                            // Fall back to default template of ContentPart, if there is not template for shape type(partTypeName)
+                            // eg. ContentPart
+                            ctx.Shape.Metadata.Alternates.Add($"ContentPart");
```

Comment:
Sorry for confusion! I copied code from my other PR #8935, I kept this here as code was related to alternates.  
In other PR #8935, I am changing shape type to dynamic part name so it would be needed later.

I'll remove if you are still not convinced.